R/kernel.r

```diff
@@ -242,6 +243,75 @@ complete = function(request) {
         cursor_end = start_position + nchar(c.info$token)))
 },
 
+inspect = function(request) {
+    # 5.0 protocol:
+    code <- request$content$code
+    cursor_pos <- request$content$cursor_pos
+
+    # Define functions to add a section to content.
+    add_new_section_plain_text <- function (content, section_name, new_content) {
+        if (is.null(new_content)) return(content)
+        title <- paste0('# ', section_name, ':\n')
+        return(paste0(content, title, new_content, '\n', sep = '\n'))
+    }
+
+    add_new_section_html <- function (content, section_name, new_content) {
+        if (is.null(new_content)) return(content)
+        title <- paste0('<h1>', section_name, ':</h1>\n')
+        return(paste0(content, title, new_content, '\n', sep = '\n'))
+    }
```

Comment:
duplicated code again. the two function only differ by their second line, so just do

```r
create_title <- list(
	'text/plain' = function(section_name) paste0('# ', section_name, ':\n'),
	'text/html'  = function(section_name) paste0('<h1>', section_name, ':</h1>\n'))
```

and put the rest of the code into `add_new_section`.